website/src/components/Chat/PluginsChooser.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  Button,
+  Text,
+  Tooltip,
+  Radio,
+  RadioGroup,
+  Box,
+  Editable,
+  EditableInput,
+  EditablePreview,
+  Image,
+  FormControl,
+  FormLabel,
+  Icon,
+  IconButton,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  Textarea,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { Cog, Edit, Plus } from "lucide-react";
+// i need
+import { AttachmentIcon, WarningIcon, CheckCircleIcon, CloseIcon } from "@chakra-ui/icons";
+import { ChangeEvent, useCallback, useEffect, useState, useRef } from "react";
+import { useTranslation } from "next-i18next";
+import { Controller, useFormContext } from "react-hook-form";
+import { ChatConfigFormData } from "src/types/Chat";
+import { useChatContext } from "./ChatContext";
+import { PluginEntry } from "src/types/Chat";
+import { get, post } from "src/lib/api";
+import { API_ROUTES } from "src/lib/routes";
+
+// TODO: Delete me!
+const measureDivWidth = (div: HTMLDivElement | null) => {
+  if (!div) return 0;
+  const width = div?.parentElement?.offsetWidth - 15;
+  return width;
+};
+
+export const PluginsChooser = ({ plugins }: { plugins: PluginEntry[] }) => {
+  const { t } = useTranslation("common");
+  const { control, register, reset, setValue } = useFormContext<ChatConfigFormData>();
+  const [selectedPluginIndex, setSelectedPluginIndex] = useState<number | null>(null);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  const findSelectedPluginIndex = useCallback(() => {
+    return plugins.findIndex((plugin) => plugin.enabled);
+  }, [plugins]);
+
+  const handlePluginSelect = useCallback(
+    (index: number) => {
+      const selectedIndex = plugins.findIndex((plugin) => plugin.enabled);
+      const newPlugins = plugins.map((plugin, i) => ({
+        ...plugin,
+        enabled: selectedIndex === index ? false : i === index,
+      }));
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      // onPluginActivated(newPlugins[index].enabled);
+    },
+    [setValue, plugins]
+  );
```

Comment:
Yes

Reply:
But also deselecting, if clicked on the already selected plugin

Reply:
this will only select one of the plugins, you might want to use a dropdown or select to manage the active plugin, instead of handling the state manually.

Reply:
Yes only one is possible to be selected, we must not allow multiple plugins